From edd27a07f3d6c5dd60e023f08dd403b021c2e1e9 Mon Sep 17 00:00:00 2001
From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Thu, 29 Sep 2022 11:05:41 +0200
Subject: [PATCH] Fix setting audacity_use_midi=off

When disabling MIDI support, cmake will fail:

```
$ cmake -B build -Daudacity_use_midi=off
[...]
CMake Error at cmake-proxies/CMakeLists.txt:314 (message):
  EXPERIMENTAL_MIDI_OUT requires USE_MIDI
```

Disable `EXPERIMENTAL_MIDI_OUT` in case MIDI support was explicitly
disabled.

Signed-off-by: Benjamin Drung <benjamin.drung@canonical.com>
---
 src/Experimental.cmake | 38 +++++++++++++++++++++-----------------
 1 file changed, 21 insertions(+), 17 deletions(-)

diff --git a/src/Experimental.cmake b/src/Experimental.cmake
index 94b5487f874..ef907e9266e 100644
--- a/src/Experimental.cmake
+++ b/src/Experimental.cmake
@@ -107,23 +107,6 @@ set( EXPERIMENTAL_OPTIONS_LIST
    # Paul Licameli (PRL) 29 Nov 2014
    #IMPROVED_SEEKING
 
-   #MIDI_IN
-
-   # RBD, 1 Sep 2008
-   # Enables MIDI Output of NoteTrack (MIDI) data during playback
-   # USE_MIDI must be defined in order for MIDI_OUT to work
-   MIDI_OUT
-
-   # JKC, 17 Aug 2017
-   # Enables the MIDI note stretching feature, which currently
-   # a) Is broken on Linux (Bug 1646)
-   # b) Crashes with Sync-Lock (Bug 1719)
-   # c) Needs UI design review.
-   #MIDI_STRETCHING
-
-   # USE_MIDI must be defined in order for SCOREALIGN to work
-   #SCOREALIGN
-
    #Automatically tries to find an acceptable input volume
    #AUTOMATED_INPUT_LEVEL_ADJUSTMENT
 
@@ -188,6 +171,27 @@ set( EXPERIMENTAL_OPTIONS_LIST
 
 # Some more flags that depend on other configuration options
 
+if( NOT audacity_use_midi STREQUAL "off" )
+   list( APPEND EXPERIMENTAL_OPTIONS_LIST
+      #MIDI_IN
+
+      # RBD, 1 Sep 2008
+      # Enables MIDI Output of NoteTrack (MIDI) data during playback
+      # USE_MIDI must be defined in order for MIDI_OUT to work
+      MIDI_OUT
+
+      # JKC, 17 Aug 2017
+      # Enables the MIDI note stretching feature, which currently
+      # a) Is broken on Linux (Bug 1646)
+      # b) Crashes with Sync-Lock (Bug 1719)
+      # c) Needs UI design review.
+      #MIDI_STRETCHING
+
+      # USE_MIDI must be defined in order for SCOREALIGN to work
+      #SCOREALIGN
+   )
+endif()
+
 if( "SCRUBBING_SUPPORT" IN_LIST EXPERIMENTAL_OPTIONS_LIST )
    list( APPEND EXPERIMENTAL_OPTIONS_LIST SCRUBBING_SCROLL_WHEEL )
 endif()
